x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/planner/AbstractPhysicalOperationProviders.java

```diff
@@ -54,6 +55,19 @@ public final PhysicalOperation groupingPhysicalOperation(
         var aggregates = aggregateExec.aggregates();
 
         var sourceLayout = source.layout;
+        AggregatorMode aggregatorMode;
+
+        if (mode == AggregateExec.Mode.FINAL) {
+            aggregatorMode = AggregatorMode.FINAL;
+        } else if (mode == AggregateExec.Mode.PARTIAL) {
+            if (aggregateExec.child() instanceof ExchangeSourceExec) {// the reducer step at data node (local) level
```

Comment:
I feel like there should be a better way to plug this in. This feels like it'd break in a sneaky way one day.

Reply:
Yes, we can add a new aggregation mode.

Reply:
👍 created https://github.com/elastic/elasticsearch/issues/108468.